c++/src/kj/exception.c++

```diff
@@ -1201,12 +1201,17 @@ InFlightExceptionIterator::InFlightExceptionIterator()
     : ptr(currentException) {}
 
 Maybe<const Exception&> InFlightExceptionIterator::next() {
-  if (ptr == nullptr) return kj::none;
+  while (ptr != nullptr) {
+    const ExceptionImpl *result = static_cast<const ExceptionImpl *>(ptr);
+    validateExceptionPointer(result);
+    ptr = result->nextCurrentException;
+    if (result->isValid()) {
```

Comment:
Maybe not in this PR since it's not new, but I think `isMovedAway()` would be a clearer name than `isValid()` (with opposite semantics).